Swift 3 updates

aldeed opened the pull request:
This is the automatic changes Xcode 8 made when converting to swift 3, plus some manual fixes to that. I don't know for sure that this all works. In general everything seems to work in our (really complex) app, but hot reload might not be working properly. It seems like it is having trouble downloading and/or refreshing to the new version, but that may or may not be related to the changes in this PR.

When I point our app to this plugin commit, it no longer needs to change anything when you choose to convert to swift 3.0. And it builds in xcode 8 immediately without errors.

I'm guessing this should be pulled into some other branch on the main repo, but opening this PR against master (the only branch) for now.

martijnwalraven commented:
Thanks again for looking into this! I created a separate [`swift3` branch](https://github.com/meteor/cordova-plugin-meteor-webapp/tree/swift3) and redid some of the changes myself in smaller steps to make sure I understood what was going on. I also made some further changes to get the tests to pass (tested on iOS 8, 9 and 10). That should give us at least some level of confidence that things are still working as expected.

I didn't really understood what you were trying to fix in your 'Fix hot reload?' commit, but that is probably something we should look into separately.

aldeed commented:
@martijnwalraven Hot reload was detecting the new version but then silently failing (basically never got as far as `onMigrate`). It was because the signature for `URLSessionDownloadDelegate` changed in iOS 10. Instead of `didBecome downloadTask: URLSessionDownloadTask` it is now `didBecome streamTask: URLSessionStreamTask`. Once I added that function with the correct signature, everything worked. So I added that with `@available` but left the old one, too.

So hot reload now works for our app except for one thing: `window.location.reload()` and `window.location.replace(window.location.href)` both crash our app. But when you reopen it, it's at the new version. So other than the pesky crash in the middle, everything is fine. I can reproduce this by just connecting through the Safari console and entering `window.location.reload()` manually. Crashes every time. The error message says it's looking for a local file named "/jobs", which does not exist. /jobs is our default route. So my only guess is that it's some problem with using react router. I am going to try doing `window.location.replace` with the path stripped off, but other than that I have no idea how to fix.

aldeed commented:
@martijnwalraven I've confirmed that the hot reload works without crashing when I do this:

```js
const urlSansPath = `${window.location.protocol}//${window.location.host}/`;
window.location.replace(urlSansPath);
```

Still kind of seems like this should be fixed in the local web app routing, but one way or another it can be fixed.

aldeed commented:
Closing this and moved the hot reload thing to a new issue: https://github.com/meteor/cordova-plugin-meteor-webapp/issues/12
